neuralpredictors/layers/cores/conv2d.py

```diff
@@ -249,6 +249,10 @@ def group_sparsity(self):
         Sparsity regularization on the filters of all the conv2d layers except the first one.
         """
         ret = 0
+
+        # return zero for depth separable convolutions
+        if not hasattr(self.features[1], "conv"):
+            return ret
```

Comment:
Could you please add a warning as well to indicate that the sparsity regularization is not applied? At the moment it is being silently ignored, which may be misleading when training models. I am not sure if this is the best place to put the warning tho. Maybe best is at initialization: check whether conv type is not `conv` and if `hidden_gamma` is not 0, then we should have a warning.

Reply:
Yes, good points! I added it to the init and printed a warning